Type error when using eslint-plugin-vitest as documented

I followed the documentation, but when using the `eslint-plugin-vitest` as documented, TypeScript reports an error as follows.
Interestingly it seems to work at runtime.

**error:**
```bash
scripts/eslint_config_test.ts:7:3 - error TS2322: Type '{ meta: { name: string; version: string; }; rules: { "prefer-lowercase-title": _typescript_eslint_utils_ts_eslint.RuleModule<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>], PluginDocs, _typescript_eslint_utils_ts_eslint.R...' is not assignable to type 'Plugin'.
  Types of property 'configs' are incompatible.
    Type '{ 'legacy-recommended': { plugins: string[]; rules: {}; }; 'legacy-all': { plugins: string[]; rules: {}; }; recommended: { name: string; plugins: { readonly vitest: any; }; rules: { readonly "vitest/expect-expect": "error"; ... 6 more ...; readonly "vitest/no-import-node-test": "error"; }; }; all: { ...; }; env: { ....' is not assignable to type 'Record<string, Config<RulesRecord> | Config<RulesRecord>[] | LegacyConfig<RulesRecord, RulesRecord>>'.
      Property 'env' is incompatible with index signature.
        Type '{ name: string; languageOptions: { globals: { suite: string; test: string; describe: string; it: string; expectTypeOf: string; assertType: string; expect: string; assert: string; chai: string; vitest: string; ... 6 more ...; onTestFinished: string; }; }; }' is not assignable to type 'Config<RulesRecord> | Config<RulesRecord>[] | LegacyConfig<RulesRecord, RulesRecord>'.
          Type '{ name: string; languageOptions: { globals: { suite: string; test: string; describe: string; it: string; expectTypeOf: string; assertType: string; expect: string; assert: string; chai: string; vitest: string; ... 6 more ...; onTestFinished: string; }; }; }' is not assignable to type 'Config<RulesRecord>'.
            The types of 'languageOptions.globals' are incompatible between these types.
              Type '{ suite: string; test: string; describe: string; it: string; expectTypeOf: string; assertType: string; expect: string; assert: string; chai: string; vitest: string; vi: string; beforeAll: string; afterAll: string; beforeEach: string; afterEach: string; onTestFailed: string; onTestFinished: string; }' is not assignable to type 'Globals'.
                Property 'suite' is incompatible with index signature.
                  Type 'string' is not assignable to type 'GlobalConf'.

7   vitest,
    ~~~~~~
````

**eslint.config.ts:**
```typescript
import vitest from '@vitest/eslint-plugin';
import {defineConfig} from 'eslint/config';

export default defineConfig({
	files: ['test/unittest/**/*.{js,ts,tsx}'],
	plugins: {
		vitest,
	},
	rules: {
		...vitest.configs.recommended.rules,
	},
});
````

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
-import type { Linter } from '@typescript-eslint/utils/ts-eslint'
+import type { Linter } from 'eslint'
 import { version } from '../package.json'
 import lowerCaseTitle, { RULE_NAME as lowerCaseTitleName } from './rules/prefer-lowercase-title'
 import maxNestedDescribe, { RULE_NAME as maxNestedDescribeName } from './rules/max-nested-describe'
@@ -255,7 +255,7 @@ const rules = {
     [preferCalledTimesName]: preferCalledTimes,
     [preferExpectTypeOfName]: preferExpectTypeOf,
     [warnTodoName]: warnTodo
-}
+} as const
 
 const plugin = {
   meta: {
@@ -330,8 +330,8 @@ const plugin = {
           onTestFinished: 'writable',
         },
       },
-    },
-  },
-}
+    } as const satisfies Linter.Config,
+  } as const,
+} as const
 
 export default plugin

